vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/ComboBoxLazyDataViewIT.java

```diff
@@ -0,0 +1,154 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.combobox.test;
+
+import com.vaadin.flow.component.combobox.testbench.ComboBoxElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.testbench.TestBenchElement;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+
+import static com.vaadin.flow.component.combobox.test.ComboBoxLazyDataViewPage.COMBO_BOX_ID;
+import static com.vaadin.flow.component.combobox.test.ComboBoxLazyDataViewPage.GET_ITEMS_BUTTON_ID;
+import static com.vaadin.flow.component.combobox.test.ComboBoxLazyDataViewPage.GET_ITEM_BUTTON_ID;
+import static com.vaadin.flow.component.combobox.test.ComboBoxLazyDataViewPage.ITEMS_LIST_ID;
+import static com.vaadin.flow.component.combobox.test.ComboBoxLazyDataViewPage.SWITCH_BUTTON_ID;
+
+@TestPath("combobox-lazy-data-view-page")
+public class ComboBoxLazyDataViewIT extends AbstractComboBoxIT {
+
+    private ComboBoxElement comboBox;
+    private TestBenchElement itemsList;
+
+    @Before
+    public void init() {
+        open();
+        waitUntil(driver -> findElements(By.tagName("vaadin-combo-box"))
+                .size() > 0);
+        comboBox = $(ComboBoxElement.class).id(COMBO_BOX_ID);
+        itemsList = $("span").id(ITEMS_LIST_ID);
+    }
+
+    @Test
+    public void getItem_withDefinedItemCountAndClientSideFilter_returnsItemFromNotFilteredSet() {
+        comboBox.setFilter("777");
+
+        waitForItems(comboBox, items -> items.size() == 1
+                && "Item 777".equals(getItemLabel(items, 0)));
+
+        clickButton(GET_ITEM_BUTTON_ID);
+
+        Assert.assertEquals("The client filter shouldn't impact the items",
+                "Item 0", itemsList.getText());
+
+        comboBox.openPopup();
+
+        waitForItems(comboBox, items -> items.size() == 1000
+                && "Item 0".equals(getItemLabel(items, 0)));
+
+        clickButton(GET_ITEM_BUTTON_ID);
+
+        Assert.assertEquals("The client filter shouldn't impact the items",
+                "Item 0", itemsList.getText());
+    }
+
+    @Test
+    public void getItem_withUnknownItemCountAndClientSideFilter_returnsItemFromNotFilteredSet() {
+        clickButton(SWITCH_BUTTON_ID);
+        comboBox.setFilter("777");
+
+        waitForItems(comboBox, items -> items.size() == 1
+                && "Item 777".equals(getItemLabel(items, 0)));
+
+        clickButton(GET_ITEM_BUTTON_ID);
+
+        Assert.assertEquals("The client filter shouldn't impact the items",
+                "Item 0", itemsList.getText());
```

Comment:
Same in this test to consider asserting `itemList` count is equal to `1000` or another way to verify the `itemList` contains all the items.

Reply:
done